In [2]:
import pandas as pd
from matplotlib import pyplot as plt

In [3]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [4]:
df=pd.read_csv("drive/My Drive/Notebooks/data/travel_tourism_dataset.csv")

In [5]:
df

Trip #no.  Duration  Cost of Travel(Entire Trip) Mode of Travel  \
0            55         1                        20111         Flight   
1           288         5                        29681            Car   
2           291         7                        24950         Flight   
3           131        19                        18676       Roadtrip   
4           281        28                        18899         Flight   
...         ...       ...                          ...            ...   
2995        142        24                        17801       Roadtrip   
2996        265        16                        23410          Train   
2997        232        22                         5656       Roadtrip   
2998        112        15                        27509            Car   
2999        283        13                        19865         Cruise   

         Stay First Name   Last Name Date of Birth  \
0       Hotel   Cristina     Fuentes    1914-06-14   
1       Hotel   Patricia       Young    1923-09-25   
2      Airbnb  Christina       Kirby    1935-01-02   
3      Airbnb    Michael      Hudson    1982-04-21   
4      Airbnb  Francisco     Hensley    1913-06-29   
...       ...        ...         ...           ...   
2995  Resorts      Kerry       Allen    2002-06-02   
2996    Hotel      Kevin     Johnson    1916-07-28   
2997    Hotel    William  Mclaughlin    1917-07-25   
2998   Airbnb     Marcus         Ray    1962-07-07   
2999   Airbnb        Tom    Fletcher    1990-03-20   

                                                Address  Age         Sex  \
0         5143 Pope Camp Apt. 028\nLake Emily, UT 71371   74  Non-Binary   
1     716 Dominguez Row Suite 775\nSouth Tiffanyboro...   19      Female   
2                      PSC 4600, Box 1237\nAPO AE 32510   36      Female   
3      7013 Bryant Club Suite 695\nSouth Tina, WI 26739   38  Non-Binary   
4        074 Scott Brook Suite 700\nLukeville, VA 46541   38  Non-Binary   
...                                                 ...  ...         ...   
2995  93471 Joyce Courts Apt. 859\nMitchellside, OK ...   73  Non-Binary   
2996  921 Katie Parkways Apt. 558\nWest Lawrence, CT...   50        Male   
2997  99384 Shirley Pine Apt. 439\nEast Angela, ND 1...   36  Non-Binary   
2998                   PSC 1488, Box 2535\nAPO AP 58905   48      Female   
2999  908 Christina Junctions Apt. 546\nPerkinsville...   22  Non-Binary   

                           Nationality Date of Travel  
0     Lao People's Democratic Republic     2022-09-13  
1                                Gabon     2021-09-04  
2                 Syrian Arab Republic     2021-01-25  
3                          Afghanistan     2022-01-28  
4                              Uruguay     2021-10-21  
...                                ...            ...  
2995                              Mali     2023-07-03  
2996                            Mexico     2022-08-22  
2997                    American Samoa     2021-02-08  
2998                            Jersey     2021-06-19  
2999                        Mozambique     2021-08-07  

[3000 rows x 13 columns]

In [6]:
df.shape

(3000, 13)

In [7]:
df.columns

Index(['Trip #no.', 'Duration', 'Cost of Travel(Entire Trip)',
       'Mode of Travel', 'Stay', 'First Name', 'Last Name', 'Date of Birth',
       'Address', 'Age', 'Sex', 'Nationality', 'Date of Travel'],
      dtype='object')

In [8]:
df['Age']

0       74
1       19
2       36
3       38
4       38
        ..
2995    73
2996    50
2997    36
2998    48
2999    22
Name: Age, Length: 3000, dtype: int64

In [9]:
df['Stay']

0         Hotel
1         Hotel
2        Airbnb
3        Airbnb
4        Airbnb
         ...   
2995    Resorts
2996      Hotel
2997      Hotel
2998     Airbnb
2999     Airbnb
Name: Stay, Length: 3000, dtype: object

In [10]:
df['Cost of Travel(Entire Trip)']

0       20111
1       29681
2       24950
3       18676
4       18899
        ...  
2995    17801
2996    23410
2997     5656
2998    27509
2999    19865
Name: Cost of Travel(Entire Trip), Length: 3000, dtype: int64

In [11]:
df['First Name']

0        Cristina
1        Patricia
2       Christina
3         Michael
4       Francisco
          ...    
2995        Kerry
2996        Kevin
2997      William
2998       Marcus
2999          Tom
Name: First Name, Length: 3000, dtype: object

In [12]:
df['Last Name']

0          Fuentes
1            Young
2            Kirby
3           Hudson
4          Hensley
           ...    
2995         Allen
2996       Johnson
2997    Mclaughlin
2998           Ray
2999      Fletcher
Name: Last Name, Length: 3000, dtype: object

In [13]:
df.loc[df['Age']<30]

Trip #no.  Duration  Cost of Travel(Entire Trip) Mode of Travel  \
1           288         5                        29681            Car   
6           173         8                        16642         Flight   
8           154        27                        25299            Car   
18            7        19                        25750       Roadtrip   
22           56        24                        10551         Flight   
...         ...       ...                          ...            ...   
2962        197        29                        18250         Flight   
2973          3        26                        19775            Car   
2976        181        18                        28957       Roadtrip   
2977          6         8                        11479            Bus   
2999        283        13                        19865         Cruise   

         Stay First Name  Last Name Date of Birth  \
1       Hotel   Patricia      Young    1923-09-25   
6      Airbnb      Debra   Mitchell    1927-02-17   
8      Airbnb    Jeffrey        Key    1913-02-13   
18      Other       Lisa      Smith    1993-09-19   
22    Resorts       John    Johnson    1922-11-04   
...       ...        ...        ...           ...   
2962    Other      Jamie     Newton    1944-12-29   
2973   Airbnb     Lauren      Lewis    2012-01-10   
2976   Airbnb        Roy  Rodriguez    1991-07-21   
2977  Resorts      Keith      Baker    1990-09-04   
2999   Airbnb        Tom   Fletcher    1990-03-20   

                                                Address  Age          Sex  \
1     716 Dominguez Row Suite 775\nSouth Tiffanyboro...   19       Female   
6      05543 Scott Mount Suite 815\nPriceland, VI 56675   25         Male   
8      4725 Ferguson Hill Suite 157\nEriktown, PR 09538   25       Female   
18                     Unit 7905 Box 2194\nDPO AE 84402   22  Transgender   
22    43152 Deborah Pine Suite 123\nWest Andreland, ...   19  Transgender   
...                                                 ...  ...          ...   
2962     0700 Heather View Apt. 797\nWeststad, IL 33725   29         Male   
2973             061 Hanson Cliff\nGuzmanfurt, NE 34866   20   Non-Binary   
2976  21399 Vasquez Estate Suite 310\nNew Maurice, M...   29   Non-Binary   
2977  32321 Wilson Landing Apt. 951\nStephensbury, A...   28   Non-Binary   
2999  908 Christina Junctions Apt. 546\nPerkinsville...   22   Non-Binary   

          Nationality Date of Travel  
1               Gabon     2021-09-04  
6          San Marino     2022-08-07  
8     Kyrgyz Republic     2022-06-28  
18             France     2021-12-15  
22            Hungary     2021-06-18  
...               ...            ...  
2962            Egypt     2020-05-22  
2973         Tanzania     2022-03-04  
2976        Mauritius     2020-11-13  
2977          Iceland     2020-10-22  
2999       Mozambique     2021-08-07  

[598 rows x 13 columns]

In [14]:
df2=df.loc[df['Age']<30]

In [15]:
df2

Trip #no.  Duration  Cost of Travel(Entire Trip) Mode of Travel  \
1           288         5                        29681            Car   
6           173         8                        16642         Flight   
8           154        27                        25299            Car   
18            7        19                        25750       Roadtrip   
22           56        24                        10551         Flight   
...         ...       ...                          ...            ...   
2962        197        29                        18250         Flight   
2973          3        26                        19775            Car   
2976        181        18                        28957       Roadtrip   
2977          6         8                        11479            Bus   
2999        283        13                        19865         Cruise   

         Stay First Name  Last Name Date of Birth  \
1       Hotel   Patricia      Young    1923-09-25   
6      Airbnb      Debra   Mitchell    1927-02-17   
8      Airbnb    Jeffrey        Key    1913-02-13   
18      Other       Lisa      Smith    1993-09-19   
22    Resorts       John    Johnson    1922-11-04   
...       ...        ...        ...           ...   
2962    Other      Jamie     Newton    1944-12-29   
2973   Airbnb     Lauren      Lewis    2012-01-10   
2976   Airbnb        Roy  Rodriguez    1991-07-21   
2977  Resorts      Keith      Baker    1990-09-04   
2999   Airbnb        Tom   Fletcher    1990-03-20   

                                                Address  Age          Sex  \
1     716 Dominguez Row Suite 775\nSouth Tiffanyboro...   19       Female   
6      05543 Scott Mount Suite 815\nPriceland, VI 56675   25         Male   
8      4725 Ferguson Hill Suite 157\nEriktown, PR 09538   25       Female   
18                     Unit 7905 Box 2194\nDPO AE 84402   22  Transgender   
22    43152 Deborah Pine Suite 123\nWest Andreland, ...   19  Transgender   
...                                                 ...  ...          ...   
2962     0700 Heather View Apt. 797\nWeststad, IL 33725   29         Male   
2973             061 Hanson Cliff\nGuzmanfurt, NE 34866   20   Non-Binary   
2976  21399 Vasquez Estate Suite 310\nNew Maurice, M...   29   Non-Binary   
2977  32321 Wilson Landing Apt. 951\nStephensbury, A...   28   Non-Binary   
2999  908 Christina Junctions Apt. 546\nPerkinsville...   22   Non-Binary   

          Nationality Date of Travel  
1               Gabon     2021-09-04  
6          San Marino     2022-08-07  
8     Kyrgyz Republic     2022-06-28  
18             France     2021-12-15  
22            Hungary     2021-06-18  
...               ...            ...  
2962            Egypt     2020-05-22  
2973         Tanzania     2022-03-04  
2976        Mauritius     2020-11-13  
2977          Iceland     2020-10-22  
2999       Mozambique     2021-08-07  

[598 rows x 13 columns]

In [18]:
df.loc[(df['Age']<30) & (df['Stay'] == "Hotel")]

Trip #no.  Duration  Cost of Travel(Entire Trip) Mode of Travel   Stay  \
1           288         5                        29681            Car  Hotel   
50          216        28                        28339            Car  Hotel   
56          164         8                        15728          Train  Hotel   
71          256        11                        17377            Bus  Hotel   
90          257        10                         5881          Train  Hotel   
...         ...       ...                          ...            ...    ...   
2810        174         1                        22963            Bus  Hotel   
2819         73        24                        13476         Cruise  Hotel   
2860        173         1                        26676            Bus  Hotel   
2866        159         1                        13803       Roadtrip  Hotel   
2899         47         3                        15401         Flight  Hotel   

       First Name Last Name Date of Birth  \
1        Patricia     Young    1923-09-25   
50         Jeremy    Conrad    2015-08-05   
56         Ronald     Ortiz    2011-10-07   
71            Amy     Dixon    2003-12-14   
90        Charles    Holden    1970-07-24   
...           ...       ...           ...   
2810  Christopher    Walker    1916-10-22   
2819         Paul      Love    2010-02-11   
2860       Joseph    Arnold    2010-08-22   
2866      Douglas  Thompson    1942-03-17   
2899       Robert   Webster    1969-05-02   

                                                Address  Age          Sex  \
1     716 Dominguez Row Suite 775\nSouth Tiffanyboro...   19       Female   
50    88823 Hill Point Apt. 589\nSouth Dakota, FM 76432   18         Male   
56                           USNS Schultz\nFPO AE 77732   22  Transgender   
71    918 Cole Extensions Apt. 064\nJohnsonville, VT...   28  Transgender   
90    02951 Christine Unions Suite 351\nSmithmouth, ...   24  Transgender   
...                                                 ...  ...          ...   
2810   2212 Miller Fork Apt. 375\nNew Brianna, CA 59566   21       Female   
2819  2726 Ortega Garden Suite 913\nChristopherburgh...   29         Male   
2860    99254 Shawn Gateway\nNew Rebeccashire, LA 92577   26         Male   
2866           819 Callahan Course\nJamiestad, PR 78512   26       Female   
2899          1163 Jacob Square\nEast Darlene, IA 48611   23  Transgender   

       Nationality Date of Travel  
1            Gabon     2021-09-04  
50     Timor-Leste     2020-01-24  
56           Malta     2022-03-14  
71       Greenland     2021-05-30  
90         Iceland     2023-08-07  
...            ...            ...  
2810      Mongolia     2023-03-18  
2819  Saint Martin     2023-08-12  
2860        Norway     2023-06-22  
2866  Saudi Arabia     2020-02-21  
2899  Saint Helena     2023-06-29  

[146 rows x 13 columns]

In [19]:
df3=df.loc[(df['Age']<30) & (df['Stay'] == "Hotel")]
df3

Trip #no.  Duration  Cost of Travel(Entire Trip) Mode of Travel   Stay  \
1           288         5                        29681            Car  Hotel   
50          216        28                        28339            Car  Hotel   
56          164         8                        15728          Train  Hotel   
71          256        11                        17377            Bus  Hotel   
90          257        10                         5881          Train  Hotel   
...         ...       ...                          ...            ...    ...   
2810        174         1                        22963            Bus  Hotel   
2819         73        24                        13476         Cruise  Hotel   
2860        173         1                        26676            Bus  Hotel   
2866        159         1                        13803       Roadtrip  Hotel   
2899         47         3                        15401         Flight  Hotel   

       First Name Last Name Date of Birth  \
1        Patricia     Young    1923-09-25   
50         Jeremy    Conrad    2015-08-05   
56         Ronald     Ortiz    2011-10-07   
71            Amy     Dixon    2003-12-14   
90        Charles    Holden    1970-07-24   
...           ...       ...           ...   
2810  Christopher    Walker    1916-10-22   
2819         Paul      Love    2010-02-11   
2860       Joseph    Arnold    2010-08-22   
2866      Douglas  Thompson    1942-03-17   
2899       Robert   Webster    1969-05-02   

                                                Address  Age          Sex  \
1     716 Dominguez Row Suite 775\nSouth Tiffanyboro...   19       Female   
50    88823 Hill Point Apt. 589\nSouth Dakota, FM 76432   18         Male   
56                           USNS Schultz\nFPO AE 77732   22  Transgender   
71    918 Cole Extensions Apt. 064\nJohnsonville, VT...   28  Transgender   
90    02951 Christine Unions Suite 351\nSmithmouth, ...   24  Transgender   
...                                                 ...  ...          ...   
2810   2212 Miller Fork Apt. 375\nNew Brianna, CA 59566   21       Female   
2819  2726 Ortega Garden Suite 913\nChristopherburgh...   29         Male   
2860    99254 Shawn Gateway\nNew Rebeccashire, LA 92577   26         Male   
2866           819 Callahan Course\nJamiestad, PR 78512   26       Female   
2899          1163 Jacob Square\nEast Darlene, IA 48611   23  Transgender   

       Nationality Date of Travel  
1            Gabon     2021-09-04  
50     Timor-Leste     2020-01-24  
56           Malta     2022-03-14  
71       Greenland     2021-05-30  
90         Iceland     2023-08-07  
...            ...            ...  
2810      Mongolia     2023-03-18  
2819  Saint Martin     2023-08-12  
2860        Norway     2023-06-22  
2866  Saudi Arabia     2020-02-21  
2899  Saint Helena     2023-06-29  

[146 rows x 13 columns]

In [22]:
df.loc[(df['Age']<30) & (df['Stay'] == "Hotel") & (df['Cost of Travel(Entire Trip)']<6000)]

Trip #no.  Duration  Cost of Travel(Entire Trip) Mode of Travel   Stay  \
90          257        10                         5881          Train  Hotel   
299         188         9                         5233            Bus  Hotel   
714         100        15                         5601            Bus  Hotel   
1638        218        24                         5744          Train  Hotel   
1932        294         3                         5648       Roadtrip  Hotel   
1936        221        25                         5631          Train  Hotel   

     First Name Last Name Date of Birth  \
90      Charles    Holden    1970-07-24   
299     Brandon  Stafford    1983-10-02   
714       Julie     Young    1958-09-11   
1638       Sean    Norman    1917-03-19   
1932    Jessica     Payne    1948-01-13   
1936   Jennifer    Miller    2012-04-23   

                                                Address  Age          Sex  \
90    02951 Christine Unions Suite 351\nSmithmouth, ...   24  Transgender   
299   90656 Ramirez Street Suite 672\nNew Craig, KY ...   27       Female   
714               3114 Le Locks\nPort Barbara, WV 13529   22       Female   
1638                   Unit 7075 Box 3564\nDPO AP 11017   20  Transgender   
1932  33945 Stephens Course Apt. 943\nAndrewfort, GA...   22       Female   
1936  9821 Brent Bridge Apt. 109\nWest Robertton, ME...   29       Female   

     Nationality Date of Travel  
90       Iceland     2023-08-07  
299    Mauritius     2021-05-07  
714      Somalia     2022-07-22  
1638      Taiwan     2021-10-06  
1932   Mauritius     2020-07-01  
1936       China     2022-09-28

In [23]:
df4=df.loc[(df['Age']<30) & (df['Stay'] == "Hotel") & (df['Cost of Travel(Entire Trip)']<6000)]

In [24]:
df4

Trip #no.  Duration  Cost of Travel(Entire Trip) Mode of Travel   Stay  \
90          257        10                         5881          Train  Hotel   
299         188         9                         5233            Bus  Hotel   
714         100        15                         5601            Bus  Hotel   
1638        218        24                         5744          Train  Hotel   
1932        294         3                         5648       Roadtrip  Hotel   
1936        221        25                         5631          Train  Hotel   

     First Name Last Name Date of Birth  \
90      Charles    Holden    1970-07-24   
299     Brandon  Stafford    1983-10-02   
714       Julie     Young    1958-09-11   
1638       Sean    Norman    1917-03-19   
1932    Jessica     Payne    1948-01-13   
1936   Jennifer    Miller    2012-04-23   

                                                Address  Age          Sex  \
90    02951 Christine Unions Suite 351\nSmithmouth, ...   24  Transgender   
299   90656 Ramirez Street Suite 672\nNew Craig, KY ...   27       Female   
714               3114 Le Locks\nPort Barbara, WV 13529   22       Female   
1638                   Unit 7075 Box 3564\nDPO AP 11017   20  Transgender   
1932  33945 Stephens Course Apt. 943\nAndrewfort, GA...   22       Female   
1936  9821 Brent Bridge Apt. 109\nWest Robertton, ME...   29       Female   

     Nationality Date of Travel  
90       Iceland     2023-08-07  
299    Mauritius     2021-05-07  
714      Somalia     2022-07-22  
1638      Taiwan     2021-10-06  
1932   Mauritius     2020-07-01  
1936       China     2022-09-28

This was so much fun! In Group A there were 598 tourists under the age of 30,in Group B there were 146 tourist who stayed in hotel and were under the age of 30, and finally in Group C there were 6 potentials and only 1 was from China and her name is **Jennifer Mille**r.